### PR TITLE
Change string 'scheme' size to uppercase value to match the scheme value

### PR DESCRIPTION
**Pull request by sallyjellyy**
thought it would be better for the safety

### CODE REVIEW

**Comment by guersam** [comment]
This is a rollback of #1107. 

Could you try `String#equalsIgnoreCase` to keep the intention of the original PR?

**Comment by sallyjellyy** [comment]
@guersam done. thanks for the recommendation!

File: zio-http/src/main/scala/zio/http/Scheme.scala [hunk 1]
```diff
@@ -59,14 +59,12 @@ object Scheme       {
...
-      else
-        scheme.length match {
-          case 5 => Scheme.HTTPS
-          case 4 => Scheme.HTTP
-          case 3 => Scheme.WSS
-          case 2 => Scheme.WS
-          case _ => null
-        }
+      else if (scheme.equalsIgnoreCase("HTTPS")) Scheme.HTTPS
+      else if (scheme.equalsIgnoreCase("HTTPS")) Scheme.HTTPS
```
Comment (guersam) [inline zio-http/src/main/scala/zio/http/Scheme.scala]:
```suggestion
```

Testing against HTTPS twice.

**Comment by jdegoes** [comment]
I think this improves the safety but at the cost of performance. Since the implications of choosing the wrong scheme are just an error, I am not sure this is any kind of security issue and I would rather keep performance.

Please re-open if you would like to continue discussing!

**Pull request closed without merging**